frontend/public/components/monitoring/alert-manager-yaml-editor.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
```

Comment:
Could the `class`es in this file be changed to functional components?

Reply:
Updated.  Wow, hadn't realized how much hooks reduce the code and make it easier to read :-)

Reply:
👍 Looks good. Thanks for doing this.